cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -78,54 +68,43 @@ public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathPa
                                       " \n" + GSON.toJson(dummy));
     }
 
-    String description = secureKeyCreateRequest.getDescription();
-    String value = secureKeyCreateRequest.getData();
-    if (Strings.isNullOrEmpty(value)) {
-      throw new BadRequestException("The data field should not be empty. This is the data that will be stored " +
-                                      "securely.");
-    }
-
-    byte[] data = value.getBytes(StandardCharsets.UTF_8);
-    secureStoreManager.put(namespace, name, data, description, secureKeyCreateRequest.getProperties());
+    secureStoreService.put(secureKeyCreateRequest, secureKeyId);
```

Comment:
does it make sense to have the ``secureKeyId`` as the first parameter? Seems more natural.